pkgs/applications/virtualization/imgcrypt/default.nix

```diff
@@ -0,0 +1,33 @@
+{ buildGoModule, fetchFromGitHub, lib }:
+buildGoModule rec {
+  pname = "imgcrypt";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "containerd";
+    repo = "imgcrypt";
+    rev = "v${version}";
+    sha256 = "177fs3p2xzwjsffcxqqllx6wi6ghfyqbvfgn95v3q7a2993yqk4k";
+  };
+
+  buildFlagsArray = [
+    "-ldflags=-X github.com/containerd/containerd/version.Version=${version}"
+  ];
+
+  vendorSha256 = null;
+  subPackages = [ "cmd/ctd-decoder" "cmd/ctr" ];
+
+  postFixup = ''
+    mv $out/bin/ctr $out/bin/ctr-enc
+  '';
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "Image encryption library and command line tool";
+    license = licenses.asl20;
+    platforms = platforms.linux;
+    inherit (src.meta) homepage;
```

Comment:
Personally I'm not a huge fan of this pattern as it requires additional effort to get to the upstream location from git.

Reply:
So you'd simply prefer being able to get the homepage url directly from derivation file? Sounds like a fair change what many would prefer to have.